### PR TITLE
Reckless: support additional installation sources

### PR DESCRIPTION
**Pull request by endothermicdev**
This adds abstractions to the search and directory traversal so that additional installation sources can be used.  It supports local git repositories, local directories, and web hosted git repos beyond github (does not require an api.)  Needs some squashing.

### CODE REVIEW

File: tools/reckless [hunk 1]
```diff
@@ -637,8 +638,9 @@ def lightning_cli(*args, timeout=15) -> dict:
         raise CLIError(out['code'], out['message'])
     if clncli.returncode == 2:
         # RPC not available - lightningd not running or using alternate config
-        err = clncli.stderr.read().decode()
+        err = clncli.stderr.decode()
         raise RPCError(err)
+    raise Exception
```
Comment (rustyrussell) [inline tools/reckless:873]: This is almost certainly wrong.... I would raise RPCError on any non-0/1 return?

**Comment by rustyrussell** [comment]
Trivial rebase on master, hoping that helps CI

**Comment by rustyrussell** [comment]
@endothermicdev needs disabling some tests on elements?

File: tools/reckless (hunk 1 above)
Comment (endothermicdev) [reply tools/reckless]: Sounds reasonable. Fixed!

**Comment by endothermicdev** [comment]
Should be okay on elements - but I did change the debug output and forgot to update the test.  Fixed that and realized there should be a test for a local install from directory source.  I can squash the new commit if you'd prefer.

**Pull request merged**